web/static.go

```diff
@@ -36,12 +36,15 @@ func (w *Web) InitStatic() {
 
 		staticHandler := staticFilesHandler(http.StripPrefix(path.Join(subpath, "static"), http.FileServer(http.Dir(staticDir))))
 		pluginHandler := staticFilesHandler(http.StripPrefix(path.Join(subpath, "static", "plugins"), http.FileServer(http.Dir(*w.ConfigService.Config().PluginSettings.ClientDirectory))))
+		pluginPublicHandler := http.StripPrefix(path.Join(subpath, "plugin"), w.NewStaticHandler(pluginPublicFiles))
 
 		if *w.ConfigService.Config().ServiceSettings.WebserverMode == "gzip" {
 			staticHandler = gziphandler.GzipHandler(staticHandler)
 			pluginHandler = gziphandler.GzipHandler(pluginHandler)
+			pluginPublicHandler = gziphandler.GzipHandler(pluginPublicHandler)
 		}
 
+		w.MainRouter.PathPrefix("/plugin/{plugin_id:[A-Za-z0-9\\_\\-\\.]+}/public/").Handler(pluginPublicHandler) //http://$SITE_URL/plugin/$PLUGIN_ID/public/{anything}
```

Comment:
Did you check, if this collides with https://github.com/mattermost/mattermost-server/blob/master/app/plugin_requests.go#L21-L41? I'm not sure how the flow of requests is.

Reply:
The handler for plugins is at /plugins (plural). This is singular. I noticed the singular reference in the thread conversation, but now I am thinking maybe I should have asked if that was intentional.

Reply:
Good catch. I missed the  `plugin` vs. `plugins` detail. But I think we should reduce confusion and stick with `plugins`. Adding a new route for this seams to much and confusing.

Reply:
Agreed. I'll make the change and test the routing priority.

Reply:
I made that change, and it did indeed clash. So I moved this to the plugin_requests (which seems like a better fit anyhow) and changed the priority there. Tests updated.